enhancements/workload-partitioning/rationalizing-configuration.md

```diff
@@ -0,0 +1,479 @@
+   implementation phase.
+
+## Proposal
+
+### API
+
+The initial implementation is limited to single-node deployments. To
+support multi-node clusters, we need to solve a couple of concurrency
+problems. The first is providing a way for the admission hook
+responsible for mutating pod definitions to reliably know when the
+feature should be enabled. The admission hook cannot store state on
+its own, so we need a new API to indicate when the feature is
+enabled.
+
+We eventually want to extend the partitioning feature to support
+non-management workloads to include workload types defined by cluster
+admins. For now, however, we want to keep the interface as simple as
+possible.  Therefore, the new API will only have status fields and
+will only be used so that the admission hook (and other consumers) can
+tell when workload partitioning is active.
+
+To manage the risk associated with rolling out a complicated feature
+like this, we are going to continue to require it to be enabled as
+part of deploying the cluster. The API does not need to support
+configuring the workload partitioning in a running cluster, so no
+controller is needed, for now.
+
+We propose that the API be owned by core OpenShift, and defined in the
+`openshift/api` repository. A separate enhancement will be written to
+describe the API in more detail.
+
+### Owner for Enablement
+
+Workload management needs two node configuration steps which overlap
+with work PAO is doing today:
+
+* Set reserved CPUs in kubelet config and enable the CPU manager
+  static policy
+* Set systemd affinity to match the reserved CPU set
+
+PAO already has logic for managing CPU pools and configuring kubelet
+and CRI-O. Since it already does a lot of what is needed to enable
+workload partitioning, we propose to extend it to handle the
+additional configuration currently being done manually by users,
+including the machine config with settings for kubelet and CRI-O.
+
+The effect of this decision is that the easiest way to use the
+workload partitioning feature will be through the
+performance-addon-operator. This means that the implementation of the
+feature is split across a few components, which avoids duplicating a
+lot of the work already done in the PAO.
+
+Workload partitioning is currently enabled as part of installing a
+cluster, and that will not change as part of the implementation of
+this design. Enabling partitioning early in this way provides
+predictable behavior from the beginning of the life-cycle, and avoids
+extra reboots required to enable it in a running cluster (an important
+consideration in environments with short, fixed maintenance
+windows). To continue to support enabling the feature during
+deployment, and to simplify the configuration, we propose to extend
+the PAO with a `render` command, similar to the one in other
+operators, to have it generate manifests for the OpenShift installer
+to consume. This avoids any need to change the OpenShift installer to
```

Comment:
I don't understand how the installer consumes these. Are you sure you write what you mean?

Reply:
"consume" may be a bad choice of words, here. I think all we mean is that they are passed as extra manifests, and the installer does what it would normally do with any extra manifests.

```suggestion
as extra manifests. This avoids any need to change the OpenShift installer to
```

Reply:
I added a note to cover that.